knowledge_graph/labelling.py

```diff
@@ -1,310 +1,443 @@
-        """Uses max- and min-timestampts to filter out the labelled passages"""
-        filtered_passages = []
-        for passage in labelled_passages:
-            passage_copy = passage.model_copy(update={"spans": []})
-            for span in passage.spans:
-                span_copy = span.model_copy(update={"labellers": [], "timestamps": []})
-                for labeller, timestamp in zip(span.labellers, span.timestamps):
-                    if self._is_between_timestamps(
-                        timestamp=timestamp,
-                        min_timestamp=min_timestamp,
-                        max_timestamp=max_timestamp,
-                    ):
-                        span_copy.labellers.append(labeller)
-                        span_copy.timestamps.append(timestamp)
-
-                if len(span_copy.labellers) > 0:
-                    passage_copy.spans.append(span_copy)
-
-            if len(passage_copy.spans) > 0:
-                filtered_passages.append(passage_copy)
-
-        return filtered_passages
-
-    def pull_labelled_passages(
+        username: str,
+        password: Optional[str] = None,
+        first_name: Optional[str] = None,
+        last_name: Optional[str] = None,
+        role: Role | str = Role.annotator,
+    ) -> User:
+        """
+        Create a new user in Argilla.
+
+        Args:
+            username: Username for the new user. Must be unique in Argilla.
+            password: Password for the new user. If not provided, a random one will be generated.
+            first_name: First name of the new user. Defaults to username if not provided.
+            last_name: Last name of the new user.
+            role: Role of the new user. Can be a Role enum or string. Options:
+                - Role.annotator or "annotator" - Can annotate records and submit responses (default)
+                - Role.admin or "admin" - Full administrative access
+                - Role.owner or "owner" - Owner role
+
+        If the user already exists, it will be returned without being re-created.
+        """
+        logger.info("Creating user: %s (role: %s)", username, role)
+
+        try:
+            user = User(
+                username=username,
+                password=password,
+                first_name=first_name or username,
+                last_name=last_name,
+                role=Role(role),
+            )
+            created_user = user.create()
+            logger.info("Successfully created user: %s", created_user.username)
+            return created_user
+        except ValueError as e:
+            error_msg = str(e).lower()
+            if "already exists" in error_msg or "unique constraint" in error_msg:
+                logger.warning(
+                    "User '%s' already exists, retrieving existing user", username
+                )
+                return self.get_user(username=username)
+            raise ValueError(f"Failed to create user '{username}'") from e
```

Comment:
see my comment on `create_workspace`. I think it's worth calling `get_user` first, and then creating a user if it doesn't exist, if you want to follow the pattern of `create_` methods also getting. But I'd suggest that create/get would be better separated, and it could then be easy to write a `create_or_get_` method that combines the two.